### ISSUE TITLE
Loading the COCO weights of YOLO to a modified yaml file

### ISSUE THREAD

**Issue by random-guest** [ultralytics/ultralytics#20526]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

load method definition from the Ultalytics reference:

This method supports loading weights from a file or directly from a weights object. It matches parameters by name and shape and transfers them to the model.

If I want to do something like this 

from ultralytics import YOLO

# Load modified yaml file, edit the backbone maybe ... call it custom_yolov11.yaml
model = YOLO("custom_yolov11.yaml")

# I want to load the COCO weights of the matched by name and shape layers only 
model = model.load("yolov11.pt")

Will the .load method skip any unmatched by name and shape layer, similar to what 

strict=False  from troch?

In other words, is the default behavior of .load from Ultalytics similar to strict=False from torch?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @random-guest, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for guidance, including examples of [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. Many common questions are already answered there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate further.

If this is a custom training ❓ Question, please include as much relevant information as possible, such as snippets of your YAML file, any custom modifications, and training logs. Also, make sure you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Feel free to join the Ultralytics community in your preferred channel! For real-time chat, check out [Discord](https://discord.com/invite/ultralytics) 🎧. For in-depth discussions, try [Discourse](https://community.ultralytics.com/). You can also connect with others on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Upgrade to the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to ensure your issue isn't already resolved:

```bash
pip install -U ultralytics
```

## Environments

YOLO runs in the following up-to-date verified environments (with dependencies like [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM ([GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/))
- **Amazon** Deep Learning AMI ([AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/))
- **Docker Image** ([Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response. An Ultralytics engineer will assist you here soon! 😊

**Comment by glenn-jocher** [comment]
Hi @random-guest,

Based on the method you're describing, you're looking at the right approach for loading weights into a modified YOLO architecture.

The Ultralytics `.load()` method does indeed match parameters by name and shape, similar to PyTorch's `strict=False` behavior. It will attempt to transfer weights where possible and skip layers that don't match, which is what you need when working with a modified backbone.

When you load weights with:
```python
model = YOLO("custom_yolov11.yaml")  # Your custom architecture
model = model.load("yolov11.pt")      # Load pretrained weights
```

The model will load compatible weights and skip incompatible ones without raising an error about missing or unexpected keys.

If you run into specific issues with your implementation, feel free to share more details about your custom architecture modifications.
